config/params/registry.go

```diff
@@ -1,10 +1,162 @@
 package params
 
+import (
+	"github.com/pkg/errors"
+	fieldparams "github.com/prysmaticlabs/prysm/config/fieldparams"
+)
+
+var ErrRegistryCollision = errors.New("registry cannot add config with conflicting fork version schedule")
+var ErrConfigNotFound = errors.New("unable to find requested BeaconChainConfig")
+var ErrCannotNullifyActive = errors.New("cannot set a config marked as active to nil")
+var ErrReplaceNilConfig = errors.New("Replace called with a nil value")
+var ErrConfigNameConflict = errors.New("config with conflicting name already exists")
+
 var Registry *registry
 
 type registry struct {
+	active        *BeaconChainConfig
+	versionToName map[[fieldparams.VersionLength]byte]string
+	nameToConfig  map[string]*BeaconChainConfig
+}
+
+func NewRegistry(configs ...*BeaconChainConfig) *registry {
```

Comment:
I reworked the package surface a fair amount, please take a look. Basically instead of exporting `Registry` with its API, I have a limited set of functions exported *at the package level* that access the shared value[1].

[1] I am now calling this a `configset`, since I realized Registry is one of those annoyingly generic names + validator registry is a thing in the language of the protocol.